plugins/expressions-converter/src/org/jetbrains/kotlinx/dataframe/ExplainerIrTransformer.kt

```diff
@@ -0,0 +1,273 @@
+                            index = 0,
+                            type = expression.type,
+                            varargElementType = null,
+                            isCrossinline = false,
+                            isNoinline = false,
+                            isHidden = false,
+                            isAssignable = false
+                        )
+                    )
+                }
+                val itSymbol = valueParameters[0].symbol
+                val source = try {
+                    source.substring(expression.startOffset, expression.endOffset)
+                } catch (e: Exception) {
+                    throw Exception("$expression ${ownerName.asString()} $source", e)
+                }
+                val expressionId = expressionId(expression)
+                val receiverId = receiver?.let { expressionId(it) }
+                val valueArguments = buildList<IrExpression?> {
+                    add(source.irConstImpl())
+                    add(ownerName.asStringStripSpecialMarkers().irConstImpl())
+                    add(IrGetValueImpl(-1, -1, itSymbol))
+                    add(expressionId.irConstImpl())
+                    add(receiverId.irConstImpl())
+                    add(data.clazz?.fqNameWhenAvailable?.asString().irConstImpl())
+                    add(data.function?.name?.asString().irConstImpl())
+                    add(IrConstImpl.int(-1, -1, pluginContext.irBuiltIns.intType, data.statementIndex))
+                }
+                body = pluginContext.irFactory.createBlockBody(-1, -1) {
+                    val callback = FqName("org.jetbrains.kotlinx.dataframe.explainer.PluginCallbackProxy.doAction")
+                    val doAction = pluginContext.referenceFunctions(callback).single()
+                    statements += IrCallImpl(
+                        startOffset = -1,
+                        endOffset = -1,
+                        type = doAction.owner.returnType,
+                        symbol = doAction,
+                        typeArgumentsCount = 0,
+                        valueArgumentsCount = valueArguments.size
+                    ).apply {
+                        val clazz = FqName("org.jetbrains.kotlinx.dataframe.explainer.PluginCallbackProxy")
+                        val plugin = pluginContext.referenceClass(clazz)!!
+                        dispatchReceiver = IrGetObjectValueImpl(-1, -1, plugin.defaultType, plugin)
+                        valueArguments.forEachIndexed { i, argument ->
+                            putValueArgument(i, argument)
+                        }
+                    }
+                }
+            }
+            val alsoLambdaExpression = IrFunctionExpressionImpl(
+                -1,
+                -1,
+                type = pluginContext.irBuiltIns.functionN(2)
+                    .typeWith(listOf(expression.type, pluginContext.irBuiltIns.unitType)),
+                alsoLambda,
+                IrStatementOrigin.LAMBDA
```

Comment:
same with trailing comma's. I think currently I'm the one adding them and you're the one leaving them out haha.

Reply:
You never know, especially with k2 around the corner. We had those incompatibility issues with korro because in 1.8.20 they suddenly added another parameter. Naming them at least helps a bit :) (Also for reviewing)